lib/files.cf

```diff
@@ -1597,17 +1597,7 @@ body perms system_owned(mode)
       mode   => "$(mode)";
       owners => { "root" };
 
-    freebsd|openbsd|netbsd|darwin::
-      groups => { "wheel" };
-
-    linux::
-      groups => { "root" };
-
-    solaris::
-      groups => { "sys" };
-
-    aix::
-      groups => { "system" };
+      groups => { "$(sys.user_data[gid])" };
```

Comment:
@nickanderson my only worry is if policy is run by non-root user, in which case a lot of changes might be queued up which cannot even be achieved due to permissions? Maybe add a guard in here to only use user_data[gid] in the case that user_data[uid] is root and/or 0?

Reply:
This is a body, and unfortunately it's not possible to add if => strcmp() to an attribute. So, instead of using sys.user_data[gid] directly, I think we would need to define something like def.system_owned_group and set that dynamicaly and then have this body reference def.system_owned_group instead.